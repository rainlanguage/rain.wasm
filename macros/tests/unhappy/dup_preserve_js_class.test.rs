#[macro_use]
extern crate wasm_bindgen_utils_macros;

struct TestStruct;

#[wasm_export]
impl TestStruct {
    #[wasm_export(preserve_js_class, preserve_js_class)]
    pub async fn some_static_method(arg: String) -> Result<String, Error> {
        Ok(String::new())
    }
}

#[wasm_export(preserve_js_class, preserve_js_class)]
pub fn some_static_method(arg: String) -> Result<String, Error> {
    Ok(String::new())
}

fn main() {}
