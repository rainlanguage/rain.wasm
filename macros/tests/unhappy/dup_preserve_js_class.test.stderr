error: duplicate `preserve_js_class` attribute
 --> tests/unhappy/dup_preserve_js_class.test.rs:8:38
  |
8 |     #[wasm_export(preserve_js_class, preserve_js_class)]
  |                                      ^^^^^^^^^^^^^^^^^

error: duplicate `preserve_js_class` attribute
  --> tests/unhappy/dup_preserve_js_class.test.rs:14:34
   |
14 | #[wasm_export(preserve_js_class, preserve_js_class)]
   |                                  ^^^^^^^^^^^^^^^^^
